### PR TITLE
Account deletion

### PR DESCRIPTION
**Pull request by danielcj2**
Connected backend to redux auth slice.

### CODE REVIEW

File: Client/src/Components/TabPanels/Account/ProfilePanel.jsx [hunk 1]
```diff
@@ -179,7 +184,12 @@ const ProfilePanel = () => {
 
   // Initiates the account deletion process
   const handleDeleteAccount = () => {
-    //TODO - implement delete account function
+    dispatch(deleteUser(authToken)).then((action) => {
+      if (action.payload.success) {
+        dispatch(clearAuthState());
+        dispatch(clearMonitorState());
+      }
+    });
```
Comment (ajhollid) [inline Client/src/Components/TabPanels/Account/ProfilePanel.jsx:192]: Can we replace this with async/await?  

```
const action = await dispatch(deleteUser(authToken))
if (action.payload.success {
    ...
}
```

I always aim for the least amount of indentaiton possible

File: Client/src/Features/Auth/authSlice.js [hunk 2]
```diff
@@ -74,6 +75,28 @@ export const update = createAsyncThunk(
   }
 );
 
+export const deleteUser = createAsyncThunk(
+  "auth/delete",
+  async (data, thunkApi) => {
+    const user = jwtDecode(data);
+
+    //1.5s delay to show loading spinner
+    await new Promise((resolve) => setTimeout(resolve, 1500));
```
Comment (ajhollid) [inline Client/src/Features/Auth/authSlice.js:84]: Is this just for demonstration purposes?  I don't think we need to slow down operations purposefully when we're ready to ship production

**Review by ajhollid — CHANGES_REQUESTED** [review]
Nice work :+1:  Unless there's an issue with using `await` I think we should use that instead of `Promise.then()` for the dispatches to keep things consistent and as readable as possible

File: Client/src/Features/Auth/authSlice.js (hunk 2 above)
Comment (danielcj2) [reply Client/src/Features/Auth/authSlice.js]: Yea pretty much, without the timeout the operation is instant. The timeout is there to tell the user _hey, a process is occurring_. I would image that with larger datasets there won't be a need for it at all.

**Review by ajhollid — APPROVED** [review]

**Pull request merged**